test/OrchardCore.Tests/Recipes/RecipeExecutorTests.cs

```diff
@@ -50,6 +51,28 @@ public async Task ShouldTrimValidScriptExpression(string recipeName, string expe
             });
         }
 
+        [Fact]
+        public async Task IncorrectTypeDefinitionImportsShouldBeBlocked()
+        {
+            var context = new BlogContext();
+            await context.InitializeAsync();
+            await context.UsingTenantScopeAsync(async scope =>
+            {
+                var recipeExecutor = scope.ServiceProvider.GetRequiredService<IRecipeExecutor>();
+                // Act
+                var executionId = Guid.NewGuid().ToString("n");
+                var recipeDescriptor = new RecipeDescriptor { RecipeFileInfo = GetRecipeFileInfo("recipe6") };
+                try
+                {
+                    await recipeExecutor.ExecuteAsync(executionId, recipeDescriptor, new Dictionary<string, object>(), CancellationToken.None);
+                }
+                catch (Exception ex)
+                {
+                    Assert.Equal("Failed to add part to ContentType 'Message', The PartName property is not allowed to be empty", ex.Message);
+                }
```

Comment:
Use `Assert.ThrowsAsync()` instead.

Reply:
Thanks, always learning something new from you!

Reply:
DONE

Reply:
I'm glad :).